### PR TITLE
feat: add signin and signout actions, refactor dashboard

### PR DESCRIPTION
**Pull request by pyphilia**
This PR adds two actions with verb `signin` and `signout` when the user signs in or out. Actions are also saved when the application automatically signs in, or when the user signs out by closing the whole app.

### CODE REVIEW

File: public/app/config/config.js [hunk 1]
```diff
@@ -79,6 +79,11 @@ const VISIBILITIES = {
   PRIVATE: 'private',
 };
 
+const DEFAULT_FORMAT = 'v1';
+const ACTIONS_VERBS = {
+  SIGNOUT: 'signout',
```
Comment (juancarlosfarah) [inline public/app/config/config.js:84]: Can you confirm this matches what Graasp does (if it does anything)? Note @ahgentil, that this is for signing out of the whole application. I.e. equivalent to signing out of Graasp.

File: public/electron.js [hunk 2]
```diff
@@ -602,6 +607,18 @@ app.on('ready', async () => {
 });
 
 app.on('window-all-closed', () => {
+  // post sign out action
```
Comment (juancarlosfarah) [inline public/electron.js:610]: Do you need to bootstrap the DB? Also, can you factor it out to its own handler?

File: src/config/constants.js [hunk 3]
```diff
@@ -114,3 +114,8 @@ export const ROWS_PER_PAGE_OPTIONS = [5, 10, 25];
 export const TABLE_ROW_HEIGHT = 43;
 
 export const DRAWER_HEADER_HEIGHT = 55;
+
+export const ACTION_VERBS = {
```
Comment (juancarlosfarah) [inline src/config/constants.js:118]: Same as above to match verbs in Graasp.

File: src/components/dashboard/ActionBarChart.js [hunk 4]
```diff
@@ -83,39 +75,57 @@ class ActionBarChart extends PureComponent {
...
+          <Bar name={t('Action Count')} dataKey="count" fill={primary[type]} />
+        </BarChart>
+      </ResponsiveContainer>
+    );
+  };
+
+  render() {
+    const { spaces, t, actions } = this.props;
+
+    if (!spaces || !actions) {
```
Comment (juancarlosfarah) [inline src/components/dashboard/ActionBarChart.js:111]: Can this lead to infinite loading for an empty desktop application? Or one that hasn't been used?

File: public/app/config/config.js (hunk 1 above)
Comment (ahgentil) [reply public/app/config/config.js:84]: The in the page view we're using:
```
  Login = 'login',
  Logout = 'logout',
```

Comment (juancarlosfarah) [reply public/app/config/config.js:84]: @ahgentil, do we do anything for the main Graasp?

Comment (ahgentil) [reply public/app/config/config.js:84]: In graasp we have "logged in" and "logged out" - in the "old" actions collection.

**Pull request merged**